src/modules/csm/csm.py

```diff
@@ -223,9 +234,35 @@ def _collect_data(self, blockstamp: BlockStamp) -> bool:
             logger.info({"msg": f"Processing checkpoint for slot {checkpoint.slot}"})
             logger.info({"msg": f"Processing {len(checkpoint.duty_epochs)} epochs"})
             checkpoint.process(blockstamp)
-        logger.info({"msg": f"All epochs processed in {time.time() - start:.2f} seconds"})
+        if checkpoints:
+            logger.info({"msg": f"All epochs processed in {time.time() - start:.2f} seconds"})
         return self.frame_performance.is_coherent
 
+    def current_frame_range(self, blockstamp: BlockStamp) -> tuple[SlotNumber, SlotNumber]:
+        l_ref_slot = self.w3.csm.get_csm_last_processing_ref_slot(blockstamp)
+        r_ref_slot = self.get_current_frame(blockstamp).ref_slot
+
+        converter = self.converter(blockstamp)
+
+        # The very first report, no previous ref slot.
+        if not l_ref_slot:
+            l_ref_slot = SlotNumber(
+                r_ref_slot - converter.get_epoch_last_slot(EpochNumber(converter.frame_config.epochs_per_frame))
+            )
```

Comment:
let's write here `todo` to redo the calculation. we had an issue for the first report (more that 7200 slots between borders)